### PR TITLE
action commands

### PR DESCRIPTION
**Pull request by Unnunu**
mostly documented and reorganized code

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.03%) on `us`.

⚠️ This PR introduces 2 warnings:
- src/battle/action_cmd.h:132: warning: redundant redeclaration of `HES_PressAButton' in same scope

File: src/battle/action_cmd.h [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+extern HudScript HES_TimingOK;
+extern HudScript HES_TimingBar1Chance;
+extern HudScript HES_TimingBar3Chances;
+extern HudScript HES_TimingCharge1;
+extern HudScript HES_TimingCharge2;
+extern HudScript HES_TimingCharge3;
+extern HudScript HES_TimingCharge4a;
+extern HudScript HES_TimingCharge4b;
+extern HudScript HES_TimingCharge4c;
+extern HudScript HES_PressAButton;
```
Comment (Fuzzy2319) [inline src/battle/action_cmd.h:132]: HES_PressAButton is already declared line 108

File: src/battle/action_cmd.h [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+extern HudScript HES_PressBButton;
```
Comment (Fuzzy2319) [inline src/battle/action_cmd.h:133]: HES_BButton  is already declared line 106

Comment (ethteck) [reply src/battle/action_cmd.h:133]: Thanks for the comments, but unnunu is aware of this via the warning tracking system we have in place

File: src/battle/action_cmd.h (hunk 1 above)
Comment (Unnunu) [reply src/battle/action_cmd.h:132]: it's HES_AButton on line 108

**Review by ethteck — APPROVED** [review]

File: include/common_structs.h [hunk 3]
```diff
@@ -863,8 +863,8 @@ typedef struct BattleStatus {
     /* 0x07D */ s8 menuStatus[4]; ///< -1 = automatically pick the first move, 0 = disabled, 1 = enabled
     /* 0x081 */ s8 actionSuccess;
     /* 0x082 */ char unk_82;
-    /* 0x083 */ s8 unk_83;
-    /* 0x084 */ s8 unk_84;
+    /* 0x083 */ s8 actionCommandMode;
```
Comment (HailSanta) [inline include/common_structs.h:866]: If only used with ActionTutorialModes, call this actionCommandTutorialMode perhaps.

File: src/battle/action_cmd.h [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    /* 0x44 */ s16 barFillLevel; // full = 10000
+    /* 0x46 */ s16 thresholdLevel;
+    /* 0x48 */ s16 barFillWidth; // X100
+    /* 0x4A */ s16 actionCommandID;
+    /* 0x4C */ s16 state;
+    /* 0x4E */ s16 prepareTime;
+    /* 0x50 */ s16 difficulty; // values from 0 to 7
+    /* 0x52 */ s16 duration;
+    /* 0x54 */ s16 frameCounter;
+    /* 0x56 */ s16 hudX;
```
Comment (HailSanta) [inline src/battle/action_cmd.h:56]: hudPosX, hudPosY here

File: src/battle/action_cmd.h [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+enum ActionTutorialModes {
+    ACTION_TUTORIAL_LEARN_BLOCK         = -1,
+    ACTION_TUTORIAL_MOVES_NOT_LEARNED   = 0,
+    ACTION_TUTORIAL_MOVES_LEARNED       = 1,
+    ACTION_TUTORIAL_GUIDED              = 2,
+    ACTION_TUTORIAL_WAIT_FOR_INPUT      = 3,
+};
+
+typedef struct ActionCommandStatus {
+    /* 0x00 */ s32 dynEntityId;
```
Comment (HailSanta) [inline src/battle/action_cmd.h:45]: You can call this workerID. Theres some confusion in the codebase about generic_entity / dynamic_entity... in the end, I think we're going to rename those to workers.

File: src/battle/action_cmd/mega_shock.c [hunk 6]
```diff
@@ -1,108 +1,99 @@
...
+    actionCommandStatus->hudPrepareTime = 30;
+    actionCommandStatus->wrongButtonPressed = FALSE;
     actionCommandStatus->barFillLevel = 0;
-    actionCommandStatus->unk_48 = 0;
-    actionCommandStatus->unk_68 = 0;
-    actionCommandStatus->hudElementX = -0x30;
-    actionCommandStatus->hudElementY = 0x50;
+    actionCommandStatus->barFillWidth = 0;
+    actionCommandStatus->isBarFilled = FALSE;
+    actionCommandStatus->hudX = -0x30;
```
Comment (HailSanta) [inline src/battle/action_cmd/mega_shock.c:29]: decimals

File: include/common_structs.h (hunk 3 above)
Comment (Unnunu) [reply include/common_structs.h:866]: After a closer look I figured out that this field is related not only to tutorial, so I changed api function names and enums back to ActionCommandMode

**Pull request merged**